src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -217,4 +224,67 @@ namespace CDASimAdapter{
         forward_message( msg , message_receiver_publisher );
     }
 
+    void CDASimConnection::populate_sensors_with_file(const std::string file_path, Json::Value& sensors_json_v){
+        //Read file from disk
+        std::ifstream in_strm;
+        in_strm.open(file_path, std::ifstream::binary);
+        if(!in_strm.is_open())
+        {
+            PLOG(logERROR) << "File cannot be opened. File path: " << file_path <<std::endl;
+            return;
+        }
+        std::string line;
+        std::stringstream ss;
+        while (std::getline(in_strm, line)) {
+            ss << line;
+        }
+        in_strm.close();
+
+       populate_json_with_string(ss.str(), sensors_json_v);
+    }
+
+    void CDASimConnection::populate_json_with_string(const std::string& json_str, Json::Value& json_v){
+        //Update JSON value with information from file
+        Json::CharReaderBuilder builder;
+        std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+        JSONCPP_STRING err;
+        if(!reader->parse(json_str.c_str(), json_str.c_str() + json_str.length(), &json_v, &err))
+        {
+            PLOG(logERROR) << "Error parsing sensors from string: " << json_str << std::endl;
+            return;
```

Comment:
What is the purpose of this return statement in a void method?

Reply:
stop the execution flow and return the control to the caller

Reply:
Looks to me that the execution is complete at this point and would return to caller anyway. Maybe if we returned a value or throw an error the caller code be notified that something went wrong. Currently logging will notify the person monitoring V2X-Hub but will not impact how the plugin performs. Do we want to do something different if this error occurs?